src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -98,16 +102,167 @@ public SecurityIndexAccessEvaluator(final Settings settings, AuditLog auditLog,
         securityDeniedActionMatcher = WildcardMatcher.from(
             restoreSecurityIndexEnabled ? securityIndexDeniedActionPatternsList : securityIndexDeniedActionPatternsListNoSnapshot
         );
+        systemIndicesAdditionalControlFlag = settings.getAsBoolean(
+            ConfigConstants.SECURITY_SYSTEM_INDICES_ADDITIONAL_CONTROL_ENABLED_KEY,
+            ConfigConstants.SECURITY_SYSTEM_INDICES_ADDITIONAL_CONTROL_ENABLED_DEFAULT
+        );
     }
 
     public PrivilegesEvaluatorResponse evaluate(
         final ActionRequest request,
         final Task task,
         final String action,
         final Resolved requestedResolved,
-        final PrivilegesEvaluatorResponse presponse
+        final PrivilegesEvaluatorResponse presponse,
+        final SecurityRoles securityRoles
     ) {
         final boolean isDebugEnabled = log.isDebugEnabled();
+
+        // As per issue #2845, the legacy access control to indices with additional protection should be kept in place in the meantime and
+        // be the default.
+        if (systemIndicesAdditionalControlFlag) {
+            evaluateNewSecuredIndicesAccess(action, requestedResolved, request, task, presponse, securityRoles, isDebugEnabled);
```

Comment:
Why `new` or `legacy` here? Can the flag be a param of the method `evaluateSystemIndicesAccess(..., permissionableSystemIndicesEnabled)`?

IMO this is a setting that users can toggle on or of. New/legacy don't really carry that much meaning and I wouldn't consider disabled permission-able system indices (the default behavior) as legacy.

Reply:
This is more a featureflag than a settings for users to turn on and off.  

Initially it was supposed to be a "flat modification", but over time  it was decided to include a flag, as we are adding additional constraints, so certain accounts could lose access to those indices.

Reply:
I am also not sure I understand why it is being separated--can you clarify Sam? I am sure you did it for a reason but I don't see why there are two methods right now.

Reply:
Thes have been combined into a single call, resolving